### PR TITLE
Added support for the Modified Crafting API

### LINKED ISSUE #160: Modified Crafting API
Add support for the Modified Crafting API, which is a new addition to the [Game Data APIs](https://develop.battle.net/documentation/world-of-warcraft/game-data-apis) for the Shadowlands expansion.  See [WoW Shadowlands API Update - Covenants, Soulbinds, & More](https://us.forums.blizzard.com/en/blizzard/t/wow-shadowlands-api-update-covenants-soulbinds-more/13385) for details about the new APIs.

### PR DESCRIPTION
**Pull request by danjagnow**
Added support for the Modified Crafting API.  Added an IModifiedCraftingApi interface to describe the capabilities of the API.  The IWarcraftClient interface implements this interface via the IGameDataApi interface.  Resolves #160.

### CODE REVIEW

File: src/ArgentPonyWarcraftClient/Models/GameDataApi/ModifiedCrafting/ModifiedCraftingReagentSlotTypeIndex.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    /// An index of Modified Crafting reagent slot types.
+    /// </summary>
+    public class ModifiedCraftingReagentSlotTypeIndex
+    {
+        /// <summary>
+        /// Gets links for the index of Modified Crafting reagent slot types.
+        /// </summary>
+        [JsonPropertyName("_links")]
+        public Links Links { get; set; }
+    }
```
Comment (willwolfram18) [inline src/ArgentPonyWarcraftClient/Models/GameDataApi/ModifiedCrafting/ModifiedCraftingReagentSlotTypeIndex.cs:15]: Looks like this is missing a `public SlotTypes { get; set; }` property
image

File: tests/ArgentPonyWarcraftClient.Tests/GameDataApi/ModifiedCraftingApiTests.cs [hunk 2]
```diff
@@ -0,0 +1,63 @@
+﻿using ArgentPonyWarcraftClient.Tests.Properties;
+using Xunit;
+
+namespace ArgentPonyWarcraftClient.Tests.GameDataApi
+{
+    public class ModifiedCraftingApiTests
+    {
+        [Fact]
+        public async void GetModifiedCraftingIndexAsync_Gets_ModifiedCraftingIndex()
```
Comment (willwolfram18) [inline tests/ArgentPonyWarcraftClient.Tests/GameDataApi/ModifiedCraftingApiTests.cs:9]: Is there a particular reason for using `async void` return type on these test methods? My understanding from [Microsoft's async/await best practices](https://docs.microsoft.com/en-us/archive/msdn-magazine/2013/march/async-await-best-practices-in-asynchronous-programming#avoid-async-void) is that it's generally preferred to use `async Task`. Granted, the perspective of that document is for application code where someone would be calling the method directly and this is a unit test, so I'm not sure if there is a specific benefit to this in a test.

**Review by willwolfram18 — CHANGES_REQUESTED** [review]

File: src/ArgentPonyWarcraftClient/Models/GameDataApi/ModifiedCrafting/ModifiedCraftingReagentSlotTypeIndex.cs (hunk 1 above)
Comment (danjagnow) [reply src/ArgentPonyWarcraftClient/Models/GameDataApi/ModifiedCrafting/ModifiedCraftingReagentSlotTypeIndex.cs:21]: I thought I had added that property, but clearly not.  Fixed now.  Good catch!

File: tests/ArgentPonyWarcraftClient.Tests/GameDataApi/ModifiedCraftingApiTests.cs (hunk 2 above)
Comment (danjagnow) [reply tests/ArgentPonyWarcraftClient.Tests/GameDataApi/ModifiedCraftingApiTests.cs]: There's no good reason to be using `async void`.  I converted these tests, and I'll do a separate PR to convert the remaining tests.  Thanks for pointing that out.

**Review by willwolfram18 — APPROVED** [review]

**Pull request merged**